programs/stake/src/stake_state.rs

```diff
@@ -1023,6 +1057,34 @@ impl MergeKind {
     }
 }
 
+/// Calculate the effective credits observed for two stakes whenn merging
+///
+/// When merging two activating or active stakes, the credits observed of the
```

Comment:
According to this definition, https://github.com/solana-labs/solana/blob/5e277897b0e33ca6030555562665b69a257dd9c7/programs/stake/src/stake_state.rs#L888-L899

we shouldn't be merging _activating_ stakes.

so, `When merging two just-delegated or fully-active stakes, ...` would be more accurate.

or `When merging two ActivationEpoch or FullyActive stakes, ...` (yeah, we lack good terminology here)

Reply:
Ah I see, I always thought `ActivationEpoch` = `activating`, but as you point out, a stake can be `activating` and also be in a transient state with some active and some activating. I worry that `just-delegated` doesn't make it clear that the stake is *only* activating. What about `purely-activating`?

Reply:
hmm, `purely-activating` sounds to me to indicate the transient activating stake account.

maybe, just using enum variant should be ok. it's just a source code comment after all?

@t-nelson do you have some word on this? (iirc, `ActivationEpoch` came from you?).